src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -3049,16 +3072,36 @@ internal Task WriteByteArray(byte[] b, int len, int offsetBuffer, bool canAccumu
                                 completion = new TaskCompletionSource<object>();
                                 task = completion.Task; // we only care about return from topmost call, so do not access Task property in other cases
                             }
-                            WriteByteArraySetupContinuation(b, len, completion, offset, packetTask);
+
+                            if (array == null)
+                            {
+                                byte[] tempArray = new byte[len];
+                                Span<byte> copyTempTo = tempArray.AsSpan();
+                                ReadOnlySpan<byte> copyTempFrom = b.Slice(remainder, len);
```

Comment:
I am confused. Why do you need b.Slice again here? `b.Slice(remainder)` was already called on line 3062

Reply:
Good question. The len needs to be there so that we only copy data not empty buffer in the case where a small part of an array is used (pool rental for example). The remainder should be changed to 0 though. I've changed it and I'm re-running the test suites.

Reply:
manual and efcore tests all pass with the push from earlier today.

Reply:
windows/native, this bit is shared though. I can force it into managed mode and test that way as well.

Reply:
OK. I will run the Unix validations.